### DISCUSSION TITLE
Adventure map, when a hero moves by revealing the adventure map, the hero's path should be adjusted to take into account the revealed terrain

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4654] [Ideas]
When a hero moves by revealing the adventure map (the Fog of War is removed as the hero moves), the hero's path should be adjusted to take into account the revealed terrain.
Currently, it is not the case neither in fheroes2 nor in the original game.
Therefore the hero loses a considerable number of movement points.

In original game:

https://user-images.githubusercontent.com/43583994/142229720-802f04b4-8137-41fd-a98d-39fd18a834f8.mp4

In fheroes2:

https://user-images.githubusercontent.com/43583994/142229872-c1f3589e-732b-485a-92d2-ff06cee77e4a.mp4

Also, in cases similar to the one presented, the shortest predefined hero path should be displayed over the Fog of War.
![Path fog of war modifié 02 001](https://user-images.githubusercontent.com/43583994/142230334-242b5146-109e-49c7-9921-0241beda83a2.png)
![Path fog of war 002](https://user-images.githubusercontent.com/43583994/142230307-a1b869bb-80ea-4f84-926b-3f6a644dfa08.png)

And if an object happens to be on this path, the path is automatically adjusted as proposed above.
![Path fog of war 03](https://user-images.githubusercontent.com/43583994/142230412-d7edee9f-4372-4f1e-9cd9-56ac869d277d.png)

Here are the save and the test scenario files:
[After eye tower 01.zip](https://github.com/ihhub/fheroes2/files/7555253/After.eye.tower.01.zip)
[After eye tower 02.zip](https://github.com/ihhub/fheroes2/files/7555855/After.eye.tower.02.zip)

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu 
I'm not sure we should allow to make a path through the fog. But at the moment hero stops it would be nice, if path is updated considering the revealed area.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> Hi, @LeHerosInconnu I'm not sure we should allow to make a path through the fog. But at the moment hero stops it would be nice, if path is updated considering the revealed area.

This would bring a fantastic convenience and gaming experience.
The player would no longer have to split the hero's path into several small paths of 3 to 4 moves (3 to 4 squares) because the fog of war is in the way. :100: :1st_place_medal:
This means hundreds of thousands of clicks less during a game. :eyeglasses: :+1:

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/10109.
I close the discussion.
